hf_vision_model_tfserving_gke/README.md

```diff
@@ -0,0 +1,103 @@
+
+There are six shell scripts to deploy 🤗 `transformers` ViT model with TF Serving on GKE. Some of them may be skipped depending on your local dev environment, but if you're starting from scratch, then run all the scripts sequentially.
+
+1. `./install_gcloud.sh`
+    - This downloads and installs `gcloud` tool, `docker-credential-gcr` and `gke-gcloud-auth-plugin` components from `gcloud` CLI. 
+  
+    - Skip this step if you already have `gcloud` installed on your system. 
+      - When you skip, and if you have not installed `docker-credential-gcr` and `gke-gcloud-auth-plugin` components yet, you should install them manually using `gcloud`.
+
+2. `./gcloud_auth.sh`
+    - This authenticates the access from `gcloud` to `GCP` project. Here are variables that you might need to customize inside the script.
+      - `GCP_PROJECT_ID`: GCP project ID to autheticate
+
+    - Skip this step if you have already authenticated beforehand.
+
+3. `./build_tfserving.sh`
+    - This builds and pushes a custom TF Serving Docker image with 🤗 `transformers` ViT model. It basically follows the [official document](https://www.tensorflow.org/tfx/serving/docker) to build an image.
+    - Here are variables that you might need to customize inside the script
+        - `GCP_PROJECT_ID`: GCP project ID to push custom built TF Serving Docker image
+        - `BASE_IMAGE_TAG`: Custom image will be built on top of this base TF Serving Docker image. This is set to `gcr.io/gcp-ml-172005/tfs-resnet-cpu-opt` by default which is CPU optimized TF Serving Docker image
+        - `MODEL_RELEASE_REPO`: GitHub repository where to download the released model. This should follow the format of `username/reponame`. This is set to `sayakpaul/deploy-hf-tf-vision-models` by default
+        - `MODEL_RELEASE_TAG`: Tag name of the GitHub Release where to download the released model. This is set to `1.0` by default
+        - `MODEL_RELEASE_FILE`: Model filename to download from the GitHub Release's Asset under `MODEL_RELEASE_REPO` and `MODEL_RELEASE_TAG`. This is set to `saved_model.tar.gz` by default
+        - `MODEL_NAME`: Model name to be exposed to client as an endpoint from TFServing. For instance, the endpoint follows the format of `http://IP_ADDRESS:PORT_NUMBER/v1/models/MODEL_NAME:predict`. This is set to `hf-vit` by default
+        - `VERSION`: Version information for the custom built TF Serving Docker image. For instance, the tag of the custom built TF Serving Docker image follows the format of `tfserving-MODEL_NAME:VERSION`. This is set to `latest` by default, so the tag will be formed as `tfserving-hf-vit:latest` by default.
+
+4. `./provision_gke_cluster.sh`
+    - This provisions a GKE clusters with `gcloud container clusters` CLI. Here are variables that you might need to customize inside the script. Most of the values in this scripts are set with the optimal values according to [this experiments](https://github.com/deep-diver/ml-deployment-k8s-tfserving)
```

Comment:
Let's also mention that the experiments were conducted with a ResNet50 model which is different from ViT.